crates/ruff_python_formatter/src/expression/string/docstring.rs

```diff
@@ -1156,6 +1188,214 @@ impl<'src> CodeExampleRst<'src> {
+                    #
+                    # NOTE: The `ticklang` and `tildlang` capture groups are
+                    # currently unused, but there was some discussion about not
+                    # assuming unlabeled blocks were Python. At the time of
+                    # writing, we do assume unlabeled blocks are Python, but
+                    # one could inspect the `ticklang` and `tildlang` capture
+                    # groups to determine whether the block is labeled or not.
+                    (?<ticks>```+)(?:\s*(?<ticklang>(?i:python|py|python3|py3))[^`]*)?
+                    |
+                    (?<tilds>~~~+)(?:\s*(?<tildlang>(?i:python|py|python3|py3))\p{any}*)?
+                )
+                $
+                ",
+            )
+            .unwrap()
+        });
+        let caps = FENCE_START.captures(rest)?;
+        let (fence_kind, fence_len) = if let Some(ticks) = caps.name("ticks") {
+            (MarkdownFenceKind::Backtick, ticks.as_str().chars().count())
+        } else {
+            let tildes = caps
+                .name("tilds")
+                .expect("no ticks means it must be tildes");
+            (MarkdownFenceKind::Tilde, tildes.as_str().chars().count())
+        };
+        Some(CodeExampleMarkdown {
+            lines: vec![],
+            opening_fence_indent: indentation_length(opening_fence_indent),
+            fence_kind,
+            fence_len,
+        })
+    }
+
+    /// Attempts to add the given line from a docstring to the Markdown code
+    /// snippet being collected.
+    ///
+    /// In this case, ownership is only not returned when the end of the block
+    /// was found, or if the block was determined to be invalid. A formatting
+    /// action is then pushed onto the queue.
+    fn add_code_line(
+        mut self,
+        original: InputDocstringLine<'src>,
+        queue: &mut VecDeque<CodeExampleAddAction<'src>>,
+    ) -> Option<CodeExampleMarkdown<'src>> {
+        if self.is_end(original) {
+            queue.push_back(self.into_format_action());
+            queue.push_back(CodeExampleAddAction::Print { original });
+            return None;
+        }
+        if !original.line.trim_whitespace().is_empty()
+            && indentation_length(original.line) < self.opening_fence_indent
+        {
+            queue.push_back(self.into_reset_action());
+            queue.push_back(CodeExampleAddAction::Print { original });
+            return None;
+        }
+        self.push(original);
+        queue.push_back(CodeExampleAddAction::Kept);
+        Some(self)
+    }
+
+    /// Returns true when given line ends this fenced code block.
+    fn is_end(&self, original: InputDocstringLine<'src>) -> bool {
+        let (_, rest) = indent_with_suffix(original.line);
+        // We can bail early if we don't have at least three backticks or
+        // tildes.
+        if !rest.starts_with("```") && !rest.starts_with("~~~") {
```

Comment:
Does `starts_with` work with over intended lines


```python
def test():
	"""
	A docstring
		over intended
		```python
		print("a")
		```
	"""
	
```

Reply:
Yeah that works, because we strip all leading whitespace.

Test case added. :-)